persist/fs/seek.go

```diff
@@ -364,10 +402,16 @@ func (s *seeker) Close() error {
 	for key := range s.indexMap {
 		delete(s.indexMap, key)
 	}
-	if s.dataFd == nil {
-		return nil
+
+	multiErr := xerrors.NewMultiError()
+	if s.dataFd != nil {
+		multiErr.Add(s.dataFd.Close())
```

Comment:
Missing `multiErr = multiErr.Add(...)` here.  

I might open an issue that we should add a metalinter for that perhaps sometime haha.

Reply:
oh wow, I didn't know that..good catch